modules/pv.py

```diff
@@ -96,32 +111,61 @@ def pv_presence_check(self, disk):
         rc, out, err = self.run_command('pvdisplay', ' ' + disk)
         ret = 0
         if self.action == 'create' and not rc:
-            self.module.exit_json(rc=0, changed= 0, msg="%s Physical Volume Exists!" % disk)
+            self.module.exit_json(rc=0, changed= 0, msg="%s Physical Volume"
+                                   " Exists!" % disk)
         elif self.action == 'remove' and rc:
-            self.module.exit_json(rc=0, changed=0,msg="%s Physical Volume Doesn't Exists!" % disk)
+            self.module.exit_json(rc=0, changed=0,msg="%s Physical Volume"
+                                   "Doesn't Exists!" % disk)
         else:
             ret = 1
         return ret
 
     def pv_action(self):
-        self.disks = self.validated_params('disks')
+        self.disks = self.module.params['disk']
         if not self.disks:
-            self.module.exit_json(msg="Nothing to do")
+            self.disks = self.module.params['disks']
+            self.module.exit_json(msg='disks paramater is deprecated, please consider using disk', rc=0, changed=True)
         return self.get_volume_command(self.disks)
 
     def get_volume_command(self, disk):
         op = 'pv' + self.action
         args = " %s %s" % (self.options, disk)
+        force = self.module.params('force')
+        if force == 'y' and self.action in ['create', 'remove']:
+            args += "-f"
+        uuid = self.module.params('uuid')
+        if uuid and self.action in ['create','pvchange']:
+            args += "-u " + uuid
+        zero = self.module.params('zero')
+        if zero and self.action == 'create':
+            args += "-Z"
+        metadatasize = self.module.params('metadatasize')
+        if metadatasize and self.action == 'create':
+            args += "--metadatasize " + metadatasize
+        metadataignore = self.module.params('metadataignore')
+        if metadataignore == 'y' and self.action in ['change']:
+            args += "--metadataignore"
+        allocatable = self.module.params('allocatable')
+        if allocatable and self.action == 'change':
+            args += "--allocatable" + allocatable
+        setphysicalvolumesize = self.module.params('setphysicalvolumesize')
+        if setphysicalvolumesize and self.action == 'resize':
+            args += "--setphysicalvolumesize" + setphysicalvolumesize
```

Comment:
In the above logic, can you group the create operations together? For example:
Only if self.action == 'create': try to see if metadatasize, zero etc are set.